apps/commercetools/src/renderDialog.ts

```diff
@@ -0,0 +1,72 @@
+// Allow lodash uniq usage. We have lodash installed anyway and using
+// the ES6 alternative (...new Set()), would require enabling downleveling
+// iteration in TS Config which with the end transpilation cost incurred
+// would be a worse solution.
+/* eslint-disable-next-line you-dont-need-lodash-underscore/uniq */
+import uniq from 'lodash/uniq';
+import get from 'lodash/get';
+import { Hash } from './interfaces';
+import { DialogExtensionSDK } from 'contentful-ui-extensions-sdk';
+
+export async function renderDialog(sdk: DialogExtensionSDK) {
+  const { projectKey, locale, clientId, clientSecret } = sdk.parameters.installation as Hash;
```

Comment:
We're still using that client secret?

Reply:
for commercetools yes, all tokens regardless of access level have both a client id and a client secret, but Ivo's picker works only with read-only tokens (you set the permissions when creating the client), admin tokens throw.